### ISSUE TITLE
3 Requests

### ISSUE THREAD

**Issue by Totesfleisch** [TheUnknownCod3r/BO4-Lucy-Menu#9]
I hope this is the right place to request Things, the Link from the Mainsite sent me here.

The first thing would be simply switching the "max Level" Feature in the Account Section with something that isnt as stat ruining, i had it twice that when opening the Menu and pressing F to enter the Sub Menu, the Menu lagged for a Split Second and registered the Input twice, setting me to max Level twice already even though i didnt want to do that.
Personally i prefer to level by myself and progress normally as thats Part of the Enjoyment for me, it would be nice if you could move that Feature further down in favor of something minor like completing the Blackmarket Contracts.

The 2nd Feature is Norecoil, i hope i didnt overlook it, that would be quite awkward.
I find the Recoil in Combination with unlimited Ammo really annoying sometimes when you stop shooting and the Weapon goes down to your Feet.

The 3rd Request is just a general Idea and i guess more wishful Thinking, but maybe there would be a Way of manipulating the Amount of Cases you could get with the Blackmarket Contracts? I remember there used to be a free Tool on another Website that let you manipulate Zombie Contracts back in Infinite Warfare Zombies and essentially you could complete a Contract infinite Times and get as many Cases as you want.

This is probably very hard or too much Work to actually do but i wanted to throw the Idea out there anyway.

In any Way, i appreciate all the Work done on this Project even if nothing from this Thread ends up being added/changed. Thank you for making Zombies more Fun :)

**Comment by TubBuddy** [comment]
your 1st request has been done it is on my channel.

your 2nd request requires to find the hashing for that which could be available  in the future.

your 3rd request i believe cant happen anymore since 3arc patched RCEs which means u need server side access to do that. Your best bet is to just use the unlock all tool by lerggy.

**Comment by Totesfleisch** [comment]
I might be blind but i cant really find anything under that name or unlock all Tools in general, mind sending me a link for that? Unless you mean the Unlock all Feature thats in this Menu already.

**Comment by TubBuddy** [comment]
https://github.com/Lerggy/t8-unlockall

**Comment by TheUnknownCod3r** [comment]
> your 1st request has been done it is on my channel.
> 
> your 2nd request requires to find the hashing for that which could be available in the future.
> 
> your 3rd request i believe cant happen anymore since 3arc patched RCEs which means u need server side access to do that. Your best bet is to just use the unlock all tool by lerggy.

Correct

No recoil would be easier than that, you shouldn't need a hash to do it. It should be simple enough to add something for it.

Correct, for Black Ops 4 and anything afterwards. Infinite warfare had a local check for crates, and you could manipulate it with Cheat Engine to alter what showed, it was fixed for BO4, so unfortunately you can't unlock it. You could Spoof it similarly to mxt for bo3, but it requires code I havent touched, so I just recommend you use the tool TubBuddy linked. Ive just merged his request so feature 1 is added, Ill add a quick change that should also fix the option pressing as soon as you open a submenu, I probably forgot a wait somewhere is all.

**Comment by TubBuddy** [comment]
I belive cheat engine closes the game

**Comment by TheUnknownCod3r** [comment]
> I belive cheat engine closes the game Ah thanks for the clarification for no recoil 👍

It does, but there's ways around that, and memory scanning still works if you suspend the process like you do on newer CoD games. MXT uses offsets, so i'd assume theres a way to do it. 

No worries. I'll see if I can get it out pretty quick.

**Comment by TubBuddy** [comment]
That would be great if you could do it asap! I would also wonder if it would be able to manipulate MP XP also? I havent checked but i dont know since MP is server side.

**Comment by TheUnknownCod3r** [comment]
I looked at it. You could do it, but I don’t have the time or patience. Strip the current menu source down to the bare minimum, remove all the functions except God Mode, Infinite Ammo and NoClip, clean up the headers and includes, and you’d maybe have a functional menu base for MP, but then you’d need to figure out how to set stats etc on top. I’m still looking to see if there’s an equivalent TableLookupIString or that function somewhere I just missed.

**Comment by TubBuddy** [comment]
Ahh ok then. Id rather just have crates tbh lol.

**Comment by TheUnknownCod3r** [comment]
> Ahh ok then, if its too much work then forget it! Id rather just have crates tbh lol.

I just published a port of the Base to Multiplayer, its tested as working, so Ill work on it in my spare time. I'd still need to look into crates, not even sure if its possible via GSC.

**Comment by Totesfleisch** [comment]
Could also update this: https://github.com/MagicMuffinBot/Magic-Muffin-Bot-Bo4-Unlock-All
Seems it worked till the recent Security Fixes on BO4. Apparently this had an MP Unlock All.

**Comment by TubBuddy** [comment]
I think with the new one u cant get all blackout skins but that gamemodes dead on pc other then that theyre both the same

**Comment by Totesfleisch** [comment]
You mean the same as the Lerggy Link you posted? Gotta be honest i couldnt find out how to inject the one you linked because theres like 15 Million Files, so i couldnt know if theyre the same.

**Comment by TubBuddy** [comment]
I used both so i know lol.

**Comment by TheUnknownCod3r** [comment]
Latest source Push adds No Recoil, it spawns a script model, and freezes you in place while firing, but thats about the best I can do at the moment. Should work for all weapons iirc.

**Comment by TubBuddy** [comment]
no recoil crashes the game and dolls still dont work

**Comment by TheUnknownCod3r** [comment]
> no recoil crashes the game and dolls still dont work

How did I even miss this... The code it decided to upload was old, and definitely didn't work which is why the crash occured. I tried a few methods before I found one that did work, and the one ive just pushed is 100% tested on all maps, and works accordingly now. I just had a couple of things in the wrong place, and one or two things missing. Thanks for that.

Matryoshka dolls are a WIP, as I stated in comments above and below the function, it was untested as I didnt have time (Im in the middle of packing stuff up to Move), once I realised it was wrong I changed it. If you'd like to help a bit faster, feel free to grab a download of the [Black Ops 4 Raw Dump](https://www.github.com/shiversoftdev/t8-src/) and scroll through the zm_orange files, in all the hashed folders too, we might find them.

**Comment by TubBuddy** [comment]
Is there a way to set game speed to half?

**Comment by TheUnknownCod3r** [comment]
Does setDvar(g_speed”,0.5); or whatever not work? I haven’t tried it.

**Comment by TubBuddy** [comment]
No it just crashes the  game

**Comment by TheUnknownCod3r** [comment]
Huh. I know it previously worked In Black Ops 3 with g_speed, as I did it that way for a dvar editor on MP. I’ll have a look at it, but the dvar should certainly be g_speed.

**Comment by TubBuddy** [comment]
I wonder if you can add a "GiveItem" function?
Eg. 
self giveItem(getitem(#"hashed"));
    self switchToItem(getitem(#"Hashed"));
I would be able to add matryoska dolls if it were true

**Comment by TheUnknownCod3r** [comment]
Not that I know of. Theres DropItem, but not GiveItem. It'll work via the GiveClientWeapon function, or Hash specific.

**Comment by TubBuddy** [comment]
G_speed only makes u go faster or slower not the game itself and fov will not go below 50 or 120. Also i would like to say that it makes more sense to put roud edit in zombies menu since its a zombies thing? Also no recoil is kinda trippy idk how u coded it but ... yeah. Also idk what pro mode does.

**Comment by TheUnknownCod3r** [comment]
Ugh, some people literally complain about everything.. Okay, lets break some stuff down. 

1: I couldn't remember the dvar values off the top of my head, so I just used random values as an example. Hence why I said in the commit notes its a POC, nothing more. 

2: It probably does, my intention is to eventually go and sort everything correctly from the host menu, but I just havent got around to it yet..

3: you say its trippy, how exactly? It freezes you in place due to linking you to a script model, thats not something that can be avoided, unless you can somehow come up with a script that does the exact same, without the script model. I guess i'll just wait for your PR to do this? I coded it the way I did because without checking if you're already linked before calling the code while not firing led to UI errors left and right, due to the entity not being linked. When you fire, it temp links you at your current coords, and when you stop it unlinks. If you can make it better, feel free thats the point of open source code.

4: Promod is just the FOV scale, it was put in by G1llam, not me. If you have an issue with that, its not my code.

**Comment by TubBuddy** [comment]
1: I was just helping.... Seeing that you didnt test it i tested it and gave feeback. 
2: sure ok i could make a PR and make changes in a few.
3: The camera goes like a 45 degree angle to the right every time you shoot. Back to point 1 that i was just helping. 
4: Ah ok makes sense.

**Comment by TheUnknownCod3r** [comment]
> 1: I was just helping.... Seeing that you didnt test it i tested it and gave feeback. 2: sure ok i could make a PR and make changes in a few. 3: The camera goes like a 45 degree angle to the right every time you shoot. Back to point 1 that i was just helping. 4: Ah ok makes sense.

Yeah I apologise, I didn't mean it to be quite as passive aggressive, I had a stressful day and directed it wrong, so my apologies. 

Yes, I messed up the values was all, it was more an example to see if they worked anyway as I hadnt tried them on BO4, so didn't matter if cg_gun_x was wrong or whatever at the time, as long as it did something.

Eventually thats the plan, I have a bunch of stuff I removed from earlier versions I wanna go back and visit, as well as obviously map functions like Catalysts, Wardens, Nosferatu's, etc. Im also wary of Menu limits, as it seems too many options eventually cause the visual glitches of the options vanishing.

Its sorta stupid a way to implement it, but due to restrictions in the injector, I couldn't do a memory edit version like you can with GSX, and im fairly sure theres not a simple dvar or function to disable recoil, so I figured as a Temp solution until I figure out something better, the script model method would work. Its tacky, it was also the way some people did it for BO2 without aimbot, but ideally i'd prefer a check for isWeaponFiring then lock the angles at a steady point, but again, I need to find the time to look at it properly, without banging my head off my desk cause the codes spitting random popup errors. 

It does? It didn't do that in testing, so like, when you fire it jerks to the right instead? I suppose I could've still messed up the entity angles, but idk why its jerking you to the right like that, so Ill have to check it I guess.

Yeah, Promod was a feature added by G1llam, back in the day it was basically a dvar that set your gun further away, making it seem like your field of view was larger than it was, but with PC its mostly irrelevant, we have that anyway. I see your PR removes it, so ill approve it since I think its pointless too, lol.

**Comment by TubBuddy** [comment]
No prob we all have one of those days haha.
And yeah it like rocks yr whole pov. I tried to see what was going on in 3rd person and the dude was spinning like one of those FaZe trickshotters lmao. ill try too also think of something for no recoil too.

**Comment by TubBuddy** [comment]
https://user-images.githubusercontent.com/127999838/230745357-1f646a20-8f92-4157-b29a-96ee350b6330.mp4

**Comment by TubBuddy** [comment]
Uploading Call of Duty Black Ops 4 2023.04.08 - 23.39.49.02(1) (online-video-cutter.com) (1).mp4…

**Comment by TubBuddy** [comment]
Uploading Call of Duty Black Ops 4 2023.04.08 - 23.39.49.02(1) (online-video-cutter.com) (1).mp4…

**Comment by TubBuddy** [comment]
https://user-images.githubusercontent.com/127999838/230745712-8294979b-012b-435b-af29-fedf32a97553.mp4

**Comment by TubBuddy** [comment]
sometime it becomes un-useable like ^

**Comment by TheUnknownCod3r** [comment]
I wonder if I just missed something off again... that literally never happened in testing, but yet, here it is. Judging from the angle tilt, i'll guess the angles are wrong, probably cause I did it like self.recoilentity.angles = (self.recoilentity.angles + self.angles); instead of just self.recoilentity.angles = self.angles;, but i'll check again.

**Comment by TheUnknownCod3r** [comment]
Can you try the latest push? a buddy of mine ran it all game while we did a long run of Voyage, and it seemed to work fine now.

**Comment by TubBuddy** [comment]
Sure ill get to that in in some time. The reason it goes like upside down us cuz i used 3rd person and no recoil and then changed it back.
I also have a question of can u have bots in blackout/spawm bots in blackout?

**Comment by TheUnknownCod3r** [comment]
No clue, I havent messed with blackout at all.

**Comment by TubBuddy** [comment]
> > I belive cheat engine closes the game Ah thanks for the clarification for no recoil 👍
> 
> It does, but there's ways around that, and memory scanning still works if you suspend the process like you do on newer CoD games. MXT uses offsets, so i'd assume theres a way to do it.
> 
> No worries. I'll see if I can get it out pretty quick.

Is there any news of this? i would love to have all the items instead of just spoofing it thnx <3

**Comment by TheUnknownCod3r** [comment]
> > > I belive cheat engine closes the game Ah thanks for the clarification for no recoil 👍
> > 
> > 
> > It does, but there's ways around that, and memory scanning still works if you suspend the process like you do on newer CoD games. MXT uses offsets, so i'd assume theres a way to do it.
> > No worries. I'll see if I can get it out pretty quick.
> 
> Is there any news of this? i would love to have all the items instead of just spoofing it thnx <3

if I can do it, it'll be MP and blackout only, and it's not looking good. Ive been going through the raw files and my memory dumps, trying to map out as many challenges and hashes as I can, and all weapon unlocks etc are server sided. The way MXT spoofed it on BO3 was almost the same as IW's method for keys, using a local client check to spoof the system. as far as im aware today now, you can only do this with an external tool of some kind, or a hash reward for giving crates, which I havent found yet. I'd assume there's a way though, as since we can give Plasma in Zombies. I'll need to look into it for a while.

**Comment by TubBuddy** [comment]
Ah Oki. I wonder if the DLL methord would work where u could make a DLL that just gives alot of crates but no skins and guns etc. So from there u can just use crates and hard unlock the items?

**Comment by TheUnknownCod3r** [comment]
That would work, there’s probably a hash for it, I just haven’t found it yet, lol. There’ll be some way to give crates in MP, and I’m probably just being stupid and missing it. Pretty sure there’s an external chest source posted by someone that you could use as a start

**Comment by TubBuddy** [comment]
No Recoil works :)
raygun bullet ammo just spits out errors left and right.
i dont think the mod vars do anything other then the player just goes faster
ill add all shields and fork

**Comment by TheUnknownCod3r** [comment]
Yeah I’ve noticed the Ray gun is messed up with certain weapons, it’s not all of them, but some of them so I’ll be removing that. Initially I tested it with the starter pistol and that’s all, which is completely on me lol.

glad no recoil works for you now, I guess it was just an angled snapping issue I did with the math, which is definitely possible. As for the mod vars, I’m adding and testing them as I go, my local version has the correct Dvars, I just need to push the commit. I noticed you added the riot shield, the stringname is just “riotshield”, so we don’t need the stupid function. I’m working on adding the map specific shields, I’ve got most of them now too I think, I just need to add them. If you check my repos, I forked a more up to date t8-src which has more of the stuff mapped, it’s helpful compared to the originals.
